lib/Importer/Caffe2.cpp

```diff
@@ -328,6 +328,19 @@ void caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     return;
   }
 
+  if (typeName == "DotProduct") {
+    // Load the inputs.
+    auto X = getNodeValueOrCreateVariableByName(op.input(0));
+    auto Y = getNodeValueOrCreateVariableByName(op.input(1));
+
+    // Create dot product node.
+    auto *node = G_.createDotProduct(opName, X, Y);
```

Comment:
cc @artemrakhov, he may have a different opinion and we should follow his recommendation. I thought we were doing as much expansion as possible in the importer, but now I am not sure seeing createReplaceNaN for instance.

Reply:
I created the PR for `ReplaceNaN` and Artem was not the one who approved it, so that may not be the pattern he recommends. Similarly, the PR for `Clip` was not approved by Artem either. It would be good to decide the proper way to do this so that we can codify it in documentation about how to add C2 and ONNX operators.

Reply:
I prefer to think about this in the following way: would you use this operator yourself if you were to create a Graph for toy example? I.e. is `createDotProduct()` alias useful to have in C++ API too, not only in loader(s)?

In my opinion, it is helpful. Probably I'd not use it for 1D inputs, but could use it for 2D tensors.